### PR TITLE
Feature-#16942 fixed holiday route bug

### PR DESCRIPTION
**Pull request by FurtunaDaniel**
remove on Employee Holidays, team leader autocomplete focus bug

### CODE REVIEW

File: app/assets/front-end/src/javascript/main.routes.js [hunk 1]
```diff
@@ -35,16 +35,16 @@ const rootTemplatePath = './views/components/';
           cssClassNames: 'view-landing'
         },
         resolve: {
-          controller: function ($location, tokenService) {
-            if(localStorage.auth_token &&
+          controller: function($location, tokenService) {
+            if (localStorage.auth_token &&
               window.location.href === 'http://localhost:8000/#/' ||
```
Comment (victorjeman) [inline app/assets/front-end/src/javascript/main.routes.js]: we should remove hardcoded links and add a config file

File: app/assets/front-end/src/javascript/components/employee/holidays/userHoliday.view.html [hunk 2]
```diff
@@ -184,19 +184,20 @@
             <div layout="column" layout-align="start start" layout-fill>
 
               <!-- SELECT TEAM LEADER -->
-              <div class="md-block u-mb-20" layout-fill>
+              <div class="md-block u-mb-20" layout-fill
+              ng-init="minLength = 0">
```
Comment (victorjeman) [inline app/assets/front-end/src/javascript/components/employee/holidays/userHoliday.view.html]: We should try to init things in controllers. Templates should have as less logic/functionality as possible.

File: app/assets/front-end/src/javascript/main.routes.js [hunk 3]
```diff
@@ -500,36 +476,17 @@ const rootTemplatePath = './views/components/';
 
 
   // Permission for EMPLOYEES ONLY
-  seeOwnProfileOnly
-    .$inject = ['tokenService', '$rootScope', 'transitionProperties'];
+  seeLimited
+    .$inject = ['$rootScope', 'tokenService'];
 
-  function seeOwnProfileOnly(tokenService, $rootScope, transitionProperties) {
+  function seeLimited($rootScope, tokenService) {
```
Comment (victorjeman) [inline app/assets/front-end/src/javascript/main.routes.js]: This function should be tested carefully

File: app/assets/front-end/src/javascript/main.routes.js [hunk 4]
```diff
@@ -500,36 +476,17 @@ const rootTemplatePath = './views/components/';
...
-      isEmployee = (decodeToken.role_id === 2);
-
-      if (isHisProfie && isEmployee) {
-        toggleMenuClassesFor('EMPLOYEE');
-        $rootScope.isAdmin = false;
-        return true;
-      } else {
-        return false;
-      }
+    if (decodeToken && decodeToken.role_id === 2) {
```
Comment (victorjeman) [inline app/assets/front-end/src/javascript/main.routes.js]: If for some reason the decodeToken is "undefined" then we will have an error on the second check

Comment (AlexandruCD) [inline app/assets/front-end/src/javascript/main.routes.js]: I don't think so. The second one will not be evaluated if decodeToken is undefined. It's short-circuiting.

Comment (victorjeman) [inline app/assets/front-end/src/javascript/main.routes.js]: yep, that seems to be the case 👍

**Pull request merged**